src/test/java/user/UserInfoTest.java

```diff
@@ -0,0 +1,280 @@
+package user;
```

Comment:
Very comprehensive testing to check if the outcome is successful, and if not the correct exceptions are throw!